cpp/src/jit/parser.cpp

```diff
@@ -339,16 +325,46 @@ std::string ptx_parser::parse_function_header(std::string const& src)
 
   auto l = std::find(f, src.cend(), ')');
 
-  auto const input_arg = parse_param_list(std::string(f, l));
-  return "\n__device__ __inline__ void " + function_name + "(" + input_arg + "){" + "\n";
+  auto const ptx_params = parse_param_list(std::string(f, l));
+
+  std::for_each(param_types.begin(), param_types.end(), [&](auto const& entry) {
+    CUDF_EXPECTS(entry.first < ptx_params.size(),
+                 "Argument index exceeds the number of parameters found in the PTX");
+  });
```

Comment:
Perhaps use `std::all_of` here instead?
```suggestion
  CUDF_EXPECTS( std::all_of( param_types.begin(), param_types.end(), [&](auto const& entry) {
    return entry.first < ptx_params.size();
  }), "Argument index exceeds the number of parameters found in the PTX");
```

Reply:
yeah, sure!